int16 id
float32 X
float32 Y
int16 flag
